docs/clients/s3.md

```diff
@@ -67,3 +67,23 @@ This information is required by AWS, and cannot be guessed by AsyncAws.
 If `ContentLength` is absent, AsyncAws will read the output before sending the
 request which could have a performance impact.
 
+
+### Download files
+```
```

Comment:
Maybe we can add a small intro telling the `GetObject` operation returns a Streamable Body that exposes 3 ways to read the content of the file.